source/grammar/qasm3.g4

```diff
@@ -370,20 +383,41 @@ additiveExpression
 
 multiplicativeExpression
     // base case either terminator or unary
-    : powerExpression
-    | unaryExpression
-    | multiplicativeExpression ( MUL | DIV | MOD ) ( powerExpression | unaryExpression )
+    : unaryExpression
+    | multiplicativeExpression ( MUL | DIV | MOD ) unaryExpression
     ;
 
 unaryExpression
-    : unaryOperator powerExpression
+    : unaryOperator? powerExpression
     ;
 
 powerExpression
+    : indexExpression
+    | powerExpression '**' indexExpression
+    ;
+
+// The general form is a comma-separated list of indexing entities.
+// `discreteSet`
+// is only valid when being used as a single index: registers can support it for
+// creating aliases, but registers cannot.
```

Comment:
The part after the comma doesn't make sense, should it be: 'but arrays cannot'?

Reply:
Ah thanks - my brain must have just given up!  Fixed in aad15aa.